src/executables/util_cs.c

```diff
@@ -4649,7 +4651,69 @@ memmon (UTIL_FUNCTION_ARG * arg)
       goto error_exit;
     }
 
+  /* execute phase */
+  error_code = mmon_get_server_info (server_info);
+  if (error_code != NO_ERROR)
+    {
+      goto error_exit;
+    }
+
+  // TODO: mmon_print_server_info (server_info);
+
+  if (module_index >= 0)
```

Comment:
Can we pare it down like this? at your choice.
```suggestion
  if (module_index >= 0)
  {
  }
  else
  {
    if (!transaction)
    {
      // default case, set the tran count
    }
    else if (show_all)
    {
     //show_all case, set the tran count 
    }
    elese
    {
     assert (false); // a parameter combination not allowed for safety 
    }
    mmon_get_module_info_summary()
  }
  
```

Reply:
I agree with removing code duplication. I change it.